docs/features/custom_logitsprocs.md

```diff
@@ -0,0 +1,142 @@
+# Custom Logits Processors
+
+This document shows you how to augment vLLM with custom logits processors.
+
+## Build a Custom Logits Processor and Pass It to the vLLM Engine
+
+Subclass `vllm.v1.sample.logits_processor.LogitsProcessor` in order to implement a custom logits processor.
+
+The contrived example below implements a custom logits processor which masks out all tokens except for one (`target_token`) with `float(-inf)`. The logits processor is disabled for any request that does not specify `target_token`.
+
+??? code "Example custom logits processor definition"
+
+    ```python
+    from typing import Optional
+    import torch
+    from vllm.config import VllmConfig
+    from vllm.sampling_params import SamplingParams
+    from vllm.v1.sample.logits_processor import (BatchUpdate,
+                                                LogitsProcessor,
+                                                MoveDirectionality)
+
+    class DummyLogitsProcessor(LogitsProcessor):
+        """Fake logit processor to support unit testing and examples"""
+
+        def __init__(self, vllm_config: "VllmConfig", device: torch.device,
+                    is_pin_memory: bool):
+            self.req_info: dict[int, SamplingParams] = {}
```

Comment:
Since we assing `self.req_info[index] = target_token`, should that type hint be `int` instead of `SamplingParams`?

```suggestion
        self.req_info: dict[int, int] = {}
```

Reply:
Thank you! 😊